src/aten/sycl/PowKernels.cpp

```diff
@@ -0,0 +1,277 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorIterator.h>
+
+#include <aten/sycl/Loops.h>
+#include <aten/sycl/UnaryKernels.h>
+
+namespace at {
+namespace native {
+namespace xpu {
+
+void pow_tensor_scalar_kernel(
+    TensorIteratorBase& iter,
+    const Scalar& exp_scalar);
+
+template <typename T>
+inline T powi(T a, T b) {
+  T result = 1;
+  while (b) {
+    if (b & 1) {
+      result *= a;
+    }
+    b /= 2;
+    a *= a;
+  }
+  return result;
+}
+
+template <typename scalar_t>
+inline scalar_t pow_impl(scalar_t a, scalar_t b) {
```

Comment:
We need align scalar behavior with CUDA.
<img width="653" alt="image" src="https://github.com/intel/torch-xpu-ops/assets/28250760/93b95be2-3956-43d6-90f6-0237e36577b9">